### PR TITLE
Better error output

### PR DESCRIPTION
**Pull request by FFY00**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -11,12 +11,14 @@ Unreleased
 - Use symlinks on creating venv's when available (`PR #274`_, Fixes `#271`_)
 - Error sooner if pip upgrade is required and fails (`PR #288`_, Fixes `#256`_)
 - Add a ``runner`` argument to ``ProjectBuilder`` (`PR #290`_, Fixes `#289`_)
+- Better error output (`PR #296`_)
```
Comment (gaborbernat) [inline CHANGELOG.rst:14]: This could be expanded a bit :thinking:

File: src/build/__init__.py [hunk 2]
```diff
@@ -332,14 +339,25 @@ def _call_backend(self, callback, outdir, config_settings=None, **kwargs):
         else:
             os.mkdir(outdir)
 
+        with self._handle_backend(callable.__name__):
+            basename = callback(outdir, config_settings, **kwargs)  # type: str
+            return os.path.join(outdir, basename)
```
Comment (gaborbernat) [inline src/build/__init__.py:344]: I think the return line might be de-indented :thinking:

**Review by gaborbernat — APPROVED** [review]

File: src/build/__init__.py (hunk 2 above)
Comment (FFY00) [reply src/build/__init__.py:344]: Eh, doesn't matter anyway, the context manager will run perfectly fine and close after the return.

Comment (FFY00) [reply src/build/__init__.py:344]: Nevermind, we want to prevent capturing exceptions from `os.path.join` 🤦

File: CHANGELOG.rst (hunk 1 above)
Comment (FFY00) [reply CHANGELOG.rst]: New message: "Hide irrelevant `pep517` error traceback and improve error messages"

File: src/build/__init__.py (hunk 2 above)
Comment (FFY00) [reply src/build/__init__.py]: Fixed.

**Review by gaborbernat — APPROVED** [review]

**Pull request merged**